client/state/plans/selectors/get-plan-prices.ts

```diff
@@ -16,36 +16,44 @@ export interface PlanPrices {
  * A convenience function that returns the undiscounted and discounted prices
  * for a given plan and site.
  */
-export function getPlanPrices(
-	state: IAppState,
-	{
-		planSlug,
-		siteId,
-		returnMonthly,
-		returnSmallestUnit,
-	}: {
-		planSlug: PlanSlug;
-		siteId: SiteId | null;
-		returnMonthly: boolean;
-		returnSmallestUnit?: boolean;
-	}
-): PlanPrices {
-	const plan = getPlan( planSlug );
-	const productId = plan?.getProductId();
+export const getPlanPrices = createSelector(
+	(
+		state: AppState,
+		{
+			planSlug,
+			siteId,
+			returnMonthly,
+			returnSmallestUnit,
+		}: {
+			planSlug: PlanSlug;
+			siteId: SiteId | null;
+			returnMonthly: boolean;
+			returnSmallestUnit?: boolean;
+		}
+	): PlanPrices => {
+		const plan = getPlan( planSlug );
+		const productId = plan?.getProductId();
 
-	return {
-		rawPrice: productId
-			? getPlanRawPrice( state, productId, returnMonthly, returnSmallestUnit )
-			: null,
-		discountedRawPrice: productId
-			? getDiscountedRawPrice( state, productId, returnMonthly, returnSmallestUnit )
-			: null,
-		planDiscountedRawPrice:
-			siteId && planSlug
-				? getPlanDiscountedRawPrice( state, siteId, planSlug, {
-						returnMonthly,
-						returnSmallestUnit,
-				  } )
+		return {
+			rawPrice: productId
+				? getPlanRawPrice( state, productId, returnMonthly, returnSmallestUnit )
 				: null,
-	};
-}
+			discountedRawPrice: productId
+				? getDiscountedRawPrice( state, productId, returnMonthly, returnSmallestUnit )
+				: null,
+			planDiscountedRawPrice:
+				siteId && planSlug
+					? getPlanDiscountedRawPrice( state, siteId, planSlug, {
+							returnMonthly,
+							returnSmallestUnit,
+					  } )
+					: null,
+		};
+	},
+	( state: AppState, { siteId } ) => [
+		getPlansBySiteId( state, siteId ?? undefined ), // consumed by getPlanDiscountedRawPrice
+		state.plans?.items, // consumed by getPlanRawPrice, getDiscountedRawPrice
+	],
+	( state: AppState, { planSlug, siteId, returnMonthly, returnSmallestUnit } ) =>
+		planSlug + siteId + returnMonthly + returnSmallestUnit
```

Comment:
This is the cache key. First time I use this, and not too fond of it living hidden here. It probably needs narrowing down too.

Not sure how else to get the dependencies array above to include non-state elements.